src/main/java/org/sagebionetworks/web/client/presenter/EmailInvitationPresenter.java

```diff
@@ -0,0 +1,158 @@
+package org.sagebionetworks.web.client.presenter;
+
+import com.google.gwt.activity.shared.AbstractActivity;
+import com.google.gwt.event.shared.EventBus;
+import com.google.gwt.query.client.Function;
+import com.google.gwt.query.client.GQuery;
+import com.google.gwt.query.client.plugins.deferred.PromiseRPC;
+import com.google.gwt.user.client.ui.AcceptsOneWidget;
+import org.sagebionetworks.repo.model.*;
+import org.sagebionetworks.web.client.GlobalApplicationState;
+import org.sagebionetworks.web.client.PlaceChanger;
+import org.sagebionetworks.web.client.SynapseClientAsync;
+import org.sagebionetworks.web.client.place.EmailInvitation;
+import org.sagebionetworks.web.client.place.LoginPlace;
+import org.sagebionetworks.web.client.place.Profile;
+import org.sagebionetworks.web.client.place.Synapse;
+import org.sagebionetworks.web.client.security.AuthenticationController;
+import org.sagebionetworks.web.client.view.EmailInvitationView;
+import org.sagebionetworks.web.client.view.users.RegisterWidget;
+import org.sagebionetworks.web.client.widget.entity.controller.SynapseAlert;
+import org.sagebionetworks.web.shared.NotificationTokenType;
+
+import javax.inject.Inject;
+
+import static org.sagebionetworks.web.client.DisplayUtils.getDisplayName;
+
+public class EmailInvitationPresenter extends AbstractActivity implements EmailInvitationView.Presenter, Presenter<EmailInvitation> {
+	private EmailInvitationView view;
+	private RegisterWidget registerWidget;
+	private SynapseClientAsync synapseClient;
+	private SynapseAlert synapseAlert;
+	private AuthenticationController authController;
+	private PlaceChanger placeChanger;
+
+	@Inject
+	public EmailInvitationPresenter(EmailInvitationView view,
+									RegisterWidget registerWidget,
+									SynapseClientAsync synapseClient,
+									SynapseAlert synapseAlert,
+									AuthenticationController authController,
+									GlobalApplicationState globalApplicationState) {
+		this.view = view;
+		this.registerWidget = registerWidget;
+		this.synapseClient = synapseClient;
+		this.synapseAlert = synapseAlert;
+		this.authController = authController;
+		this.placeChanger = globalApplicationState.getPlaceChanger();
+		view.setPresenter(this);
+		registerWidget.enableEmailAddressField(false);
```

Comment:
Initialization of view state should not be in the constructor (since state will be lost when it's reused).  Please move to init (in setPlace)

Reply:
I changed how the view is initialized, but I'd like to discuss what I did further in person.